packages/beacon-node/src/chain/stateCache/persistentCheckpointsCache.ts

```diff
@@ -602,7 +603,41 @@ export class PersistentCheckpointStateCache implements CheckpointStateCache {
   }
 
   /**
-   * Prune or persist checkpoint states in an epoch, see the description in `processState()` function
+   * Prune or persist checkpoint states in an epoch
+   * 1) If there is 1 checkpoint state with known root, persist it. This is when there is skipped slot at block 0 of epoch
+   *     slot:                           n
+   *             |-----------------------|-----------------------|
+   *     PRCS root                      |
+   *
+   * 2) If there are 2 checkpoint states, PRCS and CRCS and both roots are known to this state, persist CRCS. If the block is reorged,
+   * PRCS is regen and populated to this cache again.
+   *     slot:                           n
+   *             |-----------------------|-----------------------|
+   *     PRCS root - prune              |
+   *     CRCS root - persist             |
```

Comment:
I understand persist happens in the last 1/3 of slot 0 of epoch n. For pruning it looks like it's happening at slot 31 or epoch n-1. Is that correct?

Reply:
persist = prune from memory, and it usually happens in the last 1/3 of slot 0
to prune from disk, it happens when there is a finalized checkpoint